sdk/core/Azure.Core.Experimental/api/Azure.Core.Experimental.net461.cs

```diff
@@ -106,3 +106,26 @@ namespace Azure
         [System.Runtime.CompilerServices.MethodImpl(System.Runtime.CompilerServices.MethodImplOptions.AggressiveInlining)]public bool TryGetValue<T>(out T value) { throw null; }
     }
 }
+namespace Azure.Core.Experimental
+{
+    public partial class LruCache<TKey, TValue> : System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<TKey, TValue>>, System.Collections.IEnumerable where TKey : notnull
+    {
+        public LruCache(int capacity) { }
```

Comment:
Hmm I'm curious how we landed on adding this to the public surface. I think of it as more of something that is nice to put in Shared source but wouldn't want to publicly support it for external users.

Reply:
It came from some of the feedback I got https://github.com/Azure/azure-sdk-for-net/pull/36926#discussion_r1245428639

Reply:
I think there is hesitation about adding things to shared source since there are efforts to get rid of/ limit the number of files in there

Reply:
That is true in general, but I don't see us shipping such a low-level generic type out of Azure.Core. This is the sort of type that is perfect for shared source IMO. That said, we will definitely revisit this before shipping a GA version so we can just wait until then.